tests/e2e/validatecluster/validatecluster_test.go

```diff
@@ -189,14 +184,40 @@ var _ = Describe("Verify Create", Ordered, func() {
 			Eventually(func(g Gomega) {
 				pods, _ := e2e.ParsePods(kubeConfigFile, false)
 				count := e2e.CountOfStringInSlice("test-daemonset", pods)
-				fmt.Println("POD COUNT")
-				fmt.Println(count)
-				fmt.Println("NODE COUNT")
-				fmt.Println(len(nodes))
 				g.Expect(len(nodes)).Should((Equal(count)), "Daemonset pod count does not match node count")
 			}, "420s", "10s").Should(Succeed())
 		})
 
+		It("Verifies Restart K3s", func() {
+			_, err := e2e.DeployWorkload("daemonset.yaml", kubeConfigFile, *hardened)
+			Expect(err).NotTo(HaveOccurred(), "Daemonset manifest not deployed")
+			nodes, _ := e2e.ParseNodes(kubeConfigFile, false)
+
+			Eventually(func(g Gomega) {
+				pods, _ := e2e.ParsePods(kubeConfigFile, false)
+				count := e2e.CountOfStringInSlice("test-daemonset", pods)
+				g.Expect(len(nodes)).Should((Equal(count)), "Daemonset pod count does not match node count")
+				e2e.RestartCluster(serverNodeNames)
+				if len(agentNodeNames) > 0 {
+					e2e.RestartAgents(agentNodeNames)
+				}
+			}, "620s", "5s").Should(Succeed())
+
+			Eventually(func(g Gomega) {
+				nodes, err := e2e.ParseNodes(kubeConfigFile, false)
+				g.Expect(err).NotTo(HaveOccurred())
+				for _, node := range nodes {
+					g.Expect(node.Status).Should(Equal("Ready"))
+				}
+				cmd := "kubectl get pods -o=name -l k8s-app=test-daemonset --field-selector=status.phase=Running --kubeconfig=" + kubeConfigFile
+				res, err := e2e.RunCommand(cmd)
+				Expect(err).NotTo(HaveOccurred())
+				g.Expect(res).Should(ContainSubstring("test-daemonset"), "daemonset pods not Running")
```

Comment:
Can we change this logic to use the same functions as pre-restart?

Also, add another expect here to ensure the daemonset pods are all Running after the restart.

Reply:
Thanks but What about this block of code that is executed after the restart:
```
cmd := "kubectl get pods -o=name -l k8s-app=test-daemonset --field-selector=status.phase=Running --kubeconfig=" + kubeConfigFile
				res, err := e2e.RunCommand(cmd)
				Expect(err).NotTo(HaveOccurred())
				g.Expect(res).Should(ContainSubstring("test-daemonset"), "daemonset pods not Running")
```

it is to ensure the daemonset pods are all Running after the restart.

Reply:
That code gets the name of all pods in the `test-daemonset` app that are currently Running. 

However, it's possible that one of them is running and another one is Crashlooping, but in this case you would not see that. Also, it's possible that it's state is technically "Running", but it's stuck at "0/1 Ready". Using the same functions as earlier in the code, we can get these results and check that they are what we expect them to be